### ISSUE TITLE
Problem with file access and share mode when tailing log on windows

### ISSUE THREAD

**Issue by lhm** [elastic/filebeat#157]
Using filebeat 1.0.0-beta4 (386) / Windows Server 2012 R2.

I'm getting the following error when tailing a server log file:
```
log.go:140: ERR Failed opening c:\server\server.log: Error creating file 
'%!s(*uint16=0x12615290)': The process cannot access the file because it is being used by
another process.
```
The server log file is created with `GENERIC_WRITE` access and `FILE_SHARE_READ` share mode.

If I understand correctly, filebeat is opening the file with `GENERIC_READ | GENERIC_WRITE` access and `FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE` share mode. 
https://github.com/elastic/filebeat/blob/master/input/file_windows.go#L93-L95

When I change it in the filebeat source to use `GENERIC_READ` for access and `FILE_SHARE_READ | FILE_SHARE_WRITE` for share mode, the error disappears. This is in line with the overview this page from the MS docs: https://msdn.microsoft.com/de-de/library/windows/desktop/aa363874%28v=vs.85%29.aspx

Not sure if the above is actually the most common way of creating a server log file on windows. If there are several ways, the way that filebeat accesses the log file should maybe be configurable.

This is probably related to https://github.com/elastic/filebeat/pull/74

**Comment by ruflin** [comment]
@lhm Good timing. We also found this to be the issue. So far we only removed GENERIC_WRITE: https://github.com/elastic/filebeat/pull/156 Do you suggest to also remove FILE_SHARE_DELETE?

**Comment by lhm** [comment]
Heh, indeed. Just saw that, and it fixes the problem. And no, removing FILE_SHARE_DELETE is actually bad, it breaks the handling of file rotation. The fix in #156 is the correct one. :heart:

**Comment by ruflin** [comment]
@lhm Thanks anyways for digging into this one and also find the root cause.

As we are trying to keep track on which windows versions it worked, could you list which windows version and file system you were testing it on or running it in "production"?
